app/controllers/home_controller.rb

```diff
@@ -24,6 +24,14 @@ def post_comments
 
   def user_profile
     @user = User.find(params[:id])
+    friendship = current_user.friend_with? @user
+    if friendship.present? && friendship.status == true
+      @is_friend = true
+    elsif friendship.present? && friendship.status == nil
```

Comment:
You don't need to compare bool with `true` or `nil`
It can be simplified as: 
```ruby
@is_friend = "not_requested" and return unless friendship.present?

if friendship.status
  @is_friend = true
else
  @is_friend = false
end
```